app/org/sagebionetworks/bridge/models/schedules/ScheduleCriteria.java

```diff
@@ -5,60 +5,68 @@
 
 import org.sagebionetworks.bridge.models.Criteria;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.google.common.collect.Sets;
 
+@JsonDeserialize(builder=ScheduleCriteria.Builder.class)
 public final class ScheduleCriteria implements Criteria {
     private final Schedule schedule;
-    private final Integer minAppVersion;
-    private final Integer maxAppVersion;
-    private final Set<String> allOfGroups;
-    private final Set<String> noneOfGroups;
+    private final Criteria criteria;
```

Comment:
Looking at DDB, it's trivial to see that no one is currently using CriteriaScheduleStrategy. As such, we have a lot more leeway into changing this. Furthermore, we control all the studies currently using schedules, so it's trivially easy to ensure this remains true.

For one, if the goal is to consolidate all of the Criteria objects into a single class, it doesn't make sense for ScheduleCriteria to be a subclass of Criteria. Instead of having ScheduleCriteria contain both a Criteria and a Schedule, it would make more sense for CriteriaScheduleStrategy to instead contain a list of Criteria and a map from Criteria keys to schedules. This avoids a lot of the intermediate complexity in ScheduleCriteria.

The only complication is the integration tests, but I don't foresee that being difficult to work around.